packages/app-settings/src/index.tsx

```diff
@@ -178,9 +210,40 @@ class App extends React.PureComponent<Props, State> {
     }));
   }
 
+  private toggleCustomNode = (): void => {
+    this.setState(({ settings }: State) => ({
+      settings: {
+        ...settings,
+        /*flip the value of customNode*/
+        customNode: settings.customNode ?
+        false
+        :
+        true
+      }
+    }));
+  }
+
   private save = (): void => {
+    const { onStatusChange } = this.props;
     const { settings: { types, typesError } } = this.state;
 
+    // validate custom node url
+    const apiUrl = this.state.settings.apiUrl;
+
+    if (this.state.settings.customNode) {
```

Comment:
Are we satisfying all the of RFC 6455 "4.1 Client Requirements" https://tools.ietf.org/html/rfc6455#section-4.1, such as validating the WebSocket URIs by parsing it before connecting to it?

Are we always validating server data by safely parsing messages received on the client-side?